test/test_interface_ixxat_fd.py

```diff
@@ -0,0 +1,62 @@
+"""
+Unittest for ixxat interface using fd option.
```

Comment:
Tests are always welcome! If you want, you can also move the hardware test to `test/back2back_test.py` and let it derive from `Back2BackTestCase`. That way, a suite of generic tests are run on the device and you don't have to write any more for this interface specifically, but still get the benefits of testing. The only change would probably be to rename the `setUp()` method (see [here](https://stackoverflow.com/q/52155375/3753684)).

Reply:
I had problems with the testing I found previously (test_interface_ixxat.py).
In the begining, I had problems to run the tests. And after being able to run them, none passed.
My experience in testing in python is small (I'd like to have the oportunity to improve that), but I don't know if tests were broken, something is wrong on my environment, or I'm doing something wrong... In any case, existing tests were not testing important things (unless I'm missing something, I only saw tests checking that exceptions were triggered if some functions were called in invalid states).
So, I created test_interface_ixxat_fd.py imitating test_interface_ixxat.py (so that at least the same level of testing was achieved), and in the other hand, to ensure the code I did worked, I used a custom board I had (with personal code) that has canfd connection to talk with ixxat.
With that setup I tried simple messaging works (also with different sets of initialization options), and also more complex cases, like uds messages using udsoncan/can-isotp python libraries, to see that integration is good (I detected a bug in this scenario).
My tests are manual and only work with my board and the embedded code inside, so, I don't think we can use them in this library.
On the other side, for Back2BackTestCase based test, I would need another interface to check that works, but I have only one ixxat.
I'm open to contribute to improve testing for ixxat, but I would need some guidance (to check what's wrong with previous tests), and I would prefer to do it in a different pull request. I'm a bit worried because I saw some of the other pull requests will cause conflicts with this one and it may be hard to solve them...

Reply:
Honestly the tests are a bit buggy and not very well done for the ixxat interface. The current state is an first attempt and felix accidently broke one. So don't worry about that.

Reply:
Hey @fjburgos that's alright, then we merge this without thorough testing. And sorry, I indeed broke a test which I didn't realize since the CI doesn't run that test. Thanks @marcel-kanter for caching that.

In the end, testing is about finding bugs. If you tested with a manual setup, that's fine (for now) too! If you are interested in more testing, just open an new issue and we can discuss it there. 😄